monitoring_tools.py

```diff
@@ -0,0 +1,220 @@
+        print(f"Email sent to {recipient} with subject: {subject}")
+    except subprocess.CalledProcessError as e:
+        print(f"Error sending email: {e}")
+
+
+def send_out_emails(subject, config):
+    """standard function for looping through mailing list in config file"""
+    for email in config.mailing_list:
+        send_txt_to_email(subject=subject, recipient=email)
+
+
+def clear_txt(path):
+    """clears out the all the contents of a .txt file , leaving a blank file"""
+    with open(path, 'w') as file:
+        pass
+
+
+def add_imagepath_to_txt(path, barcode, success):
+    """add_filepath_to_monitor_txt: adds single line to end of txt file,
+        in this case with 4 spaces,
+        to keep alignment with generic template
+        args:
+            path: path to the txt file
+            failure: indicates whether image at filepath failed to upload to image server or not
+    """
+    monitor_line = f"<tr style='width: 50%'><td>{path}</td> <td>{barcode}</td><td>{success}</td></tr>"
+
+    with open("import_monitoring.html", "r") as file:
+        html_content = file.readlines()
+
+    insert_position = len(html_content) - 6
+
+    html_content.insert(insert_position, monitor_line + '\n')
+
+    # Write the updated HTML content back to the file
+    with open("import_monitoring.html", 'w') as file:
+        file.writelines(html_content)
+
+
+def add_line_between(txt_file, line_num, string):
+    """add_line_between: used to add a string line into a txt file, between two existing lines,
+       using a line index.
+       args:
+            txt_file: path to txt file to read
+            line_num: the line number after which to insert a new line/lines of text
+            string: the actual line of text you wish to insert."""
+    with open(txt_file, "r") as file:
+        lines = file.readlines()
+
+    lines.insert(line_num, string + "\n")
+
+    with open(txt_file, "w") as file:
+        file.writelines(lines)
+
+
+def create_summary_term_list(value_list, config):
+    if value_list is None:
+        return None
+    else:
+        terms = ""
+        for index, term in enumerate(config.SUMMARY_TERMS):
+            terms += f"<li>{term}: {value_list[index]}</li>"
+        return terms
+
+def add_format_batch_report(num_records, uploader, md5_code, custom_terms):
+    """add_format_batch_report:
+        creates template of upload batch report. Takes standard summary terms and args,
+       and allows for custom terms to be added with custom_terms.
+       args:
+            num_records: the number of records uploaded to DB
+            uploader: agent id , or name of uploader.
+            md5_code: the md5 code of this current upload batch.
+            config: the config file to use.
+            custom_terms: the list of custom values to add as summary terms, myst correspond with order of
+                          SUMMARY_TERMS variable in config."""
+    if custom_terms is None:
+        custom_terms = ""
+    else:
```

Comment:
this can live in an external file. Check out templating tools - jinja is the classic, but I don't it plays well with python. not critical for commit. templating tools might be overkill for this, but if you can find a lightweight one, that isn't a bad idea.